### PR TITLE
sbcl: Increase default heap size to 3GB.

### PR DESCRIPTION
**Pull request by aadcg**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/compilers/sbcl/default.nix [hunk 1]
```diff
@@ -173,6 +173,7 @@ stdenv.mkDerivation (self: rec {
 
   buildArgs = [
     "--prefix=$out"
+    "--dynamic-space-size=3072"
```
Comment (AndersonTorres) [inline pkgs/development/compilers/sbcl/default.nix:176]: 1. Why? What is the reason behind changing the default?
2. Make it configurable. Something like an argument `dynamicSpaceSize ? 3 * 1024`, plus something like

```suggestion
    "--dynamic-space-size=${dynamicSpaceSize}"
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/compilers/sbcl/default.nix (hunk 1 above)
Comment (7c6f434c) [reply pkgs/development/compilers/sbcl/default.nix:176]: It depends on how many SBCL's you want to launch at the same time (and is it nicer or not to skip the extra argument), so I'd say that if we want to override a default without a clear justification for an exact number, we should make it configurable.

**Review by hraban — CHANGES_REQUESTED** [review]
I am not a fan of changing the default [edit: in nixpkgs]. This should be taken up with SBCL devs, it's not for us to decide. They say 1GiB, we do 1GiB.

I do like the idea of allowing this to be more easily configured, but it should probably be a setuphook and a wrapper to do so, just as how nix controls e.g. GCC.

**Comment by hraban** [comment]
> After taking all views into account, I believe that the best move is to close the PR. Thanks.

Thanks for your PR though, and maybe it helps to know I constantly run into the 1GiB heap limit as well. Allowing for controlling the heap size through a setup hook has been on my todo list for ages so I completely empathize.

If you manage to convince upstream SBCL to change the default you have my +1.

**Comment by hraban** [comment]
For the record: SBCL from Nix can now have its memory controlled using an envvar:

```
$ sbcl --script <<<'(format T "~A~%" (sb-kernel::dynamic-space-size))'
1073741824
$ NIX_SBCL_DYNAMIC_SPACE_SIZE=3gb sbcl --script <<<'(format T "~A~%" (sb-kernel::dynamic-space-size))'
3221225472
```

**Pull request closed without merging**